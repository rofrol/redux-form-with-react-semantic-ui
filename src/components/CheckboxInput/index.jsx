import React from 'react';
import { Checkbox } from 'semantic-ui-react';

const CheckboxInput = (props) => {
  const { disabled, label, name, checkboxValue, defaultChecked, checked, onClick, value, input } = props;
  console.log(props);
  let checkboxProps;
  if (input) {
    checkboxProps = {
      label,
      onChange: (event, {checked}) => input.onChange(checked),
      checked: input.value ? true : false,
    };
  } else {
    checkboxProps = {
      label,
      checked,
      onClick,
    };
  }
  return (
    <div className="checkbox-wrapper">
      <Checkbox
        disabled={disabled}
        label={label}
        name={name}
        {...checkboxProps}
      />
    </div>
  );
};

CheckboxInput.defaultProps = {
  defaultChecked: false,
};

export default CheckboxInput;
