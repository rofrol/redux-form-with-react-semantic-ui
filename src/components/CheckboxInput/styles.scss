@import "~styles/variables";

.checkbox-wrapper {
  position: relative;
  width: 100%;

  .ui.checkbox {

    input {
      margin-top: 3px;
    }

    label {
      color: $color-outer-space;
      font-size: 14px;
      line-height: 24px;

      &::before {
        margin-top: 3px;
        background-color: transparent;
        border-radius: 3px;
        border: 2px solid $color-ocean-green;
      }

      &::after {
        margin-top: 3px;
        line-height: 18px;
      }
    }
  }

  .ui.checkbox input:checked~label:before {
    background-color: transparent;
    border-color: $color-ocean-green;
  }

  .ui.checkbox input:focus~.box:before, .ui.checkbox input:focus~label:before {
    background-color: transparent;
    border-color: $color-ocean-green;
  }

  .ui.checkbox input:checked~label:after {
    background-color: $color-ocean-green;
    border-radius: 3px;
    color: $color-white;
  }

  .ui.radio.checkbox input:focus:checked~label:after {
    background-color: $color-ocean-green;
    border-radius: 3px;
    color: $color-white;
  }

  .ui.checkbox .box:hover::before,
  .ui.checkbox label:hover::before {
    background-color: transparent;
    border-color: $color-ocean-green;
  }

  .ui.checkbox input:checked:focus~.box:after,
  .ui.checkbox input:checked:focus~label:after,
  .ui.checkbox input:not([type=radio]):indeterminate:focus~.box:after,
  .ui.checkbox input:not([type=radio]):indeterminate:focus~label:after {
    color: $color-white;
  }
}
