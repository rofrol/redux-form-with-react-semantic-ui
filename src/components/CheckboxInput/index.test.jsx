import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Checkbox } from 'semantic-ui-react';
import renderer from 'react-test-renderer';
import CheckboxInput from './index';

describe('Checkbox Input', () => {
  let input;
  let onChange;
  beforeEach(() => {
    onChange = jest.fn();
    input = shallow(
      <CheckboxInput
        label="some label"
        name="some name"
        value="some value"
        checked
        onChange={onChange}
      />);
  });

  const checkBox = renderer.create(<Checkbox
    label="some label"
    name="some name"
    value="some value"
    checked={false}
    onChange={onChange}
  />).toJSON();

  it('renders Checkbox Input with Checkbox', () => {
    expect(input.find(Checkbox).length).toBe(1);
  });

  it('randers Checkbox correctly', () => {
    expect(checkBox).toMatchSnapshot();
  });
});
