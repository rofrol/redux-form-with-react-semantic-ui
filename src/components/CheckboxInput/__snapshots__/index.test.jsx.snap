// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Input randers Checkbox correctly 1`] = `
<div
  className="ui checkbox"
  onChange={[Function]}
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <input
    checked={false}
    className="hidden"
    name="some name"
    readOnly={true}
    tabIndex={0}
    type="checkbox"
    value="some value"
  />
  <label>
    some label
  </label>
</div>
`;
